cloudevents/working-drafts/minor-versions.md

```diff
@@ -0,0 +1,337 @@
+past - in other words, a graduation process from "suggested as useful"
+to "officially specified". However, there's no requirement for an
+attribute to be a documented extension attribute before being added;
+that's just an informal part of the usual process.
+
+SemVer minor versions are intended for new features that don't break
+existing code. How SemVer applies to specifications is less clear-cut
+than how it applies to actual code. This document attempts to imagine
+the impact of a change on various stakeholders.
+
+For the purpose of this document, we'll consider a single new optional
+attribute into version 1.1:
+
+- Name: `xyz`
+- Type: integer
+- Constraints: value in the range 1-100 inclusive
+
+Obviously this is entirely artificial, but it should be enough to help
+find corner cases and problems.
+
+## Logical compatibility
+
+Every valid v1.1 event can be "rewritten" as a valid v1.0 event, but the reverse is not true.
+A v1.0 event containing an `xyz` extension attribute with value "test" is valid,
+but cannot be expressed as a valid v1.1 event. As there are no restrictions on
+v1.0 extension attributes, there's no way that the v1.1 specification can *claim*
+that all v1.0 events would be valid as v1.1 events, even if the addition
+is only a previously-documented extension attribute. (Extension attribute documentation
+is advisory, not mandatory.)
+
+## Event producer
+
+An event producer that doesn't know about v1.1 should create an event
+with v1.0. If the code producing the event uses an SDK, then it's *possible*
+that it will create a v1.1 event accidentally after an SDK upgrade - and that
+could cause problems. (The SDK handling of minor versions is relatively
+complex, and discussed later.) But if they produce a v1.0 event with an
+`xyz` attribute with value "test", that's fine: it's still a valid v1.0 event.
+
+An event producer that is aware of v1.1 should ensure that any use of the `xyz`
+attribute is valid - in other words, it can't have values of "test"
+(wrong type) or 123 (constraint violation). That would be an invalid v1.1 event
+in the same way that specifying a `time` attribute value of "lunchtime" would be
+invalid.
+
+## Event proxies
+
+Proxies which are unaware of v1.1 should be able to proxy v1.1 events without
+any issues, *unless* they attach special meaning to unspecified attributes (whether
+`xyz` or anything else). For example, a proxy which makes use of the `sequence` extension
+attribute but isn't aware of v1.0 cannot safely assume that v1.1 events will use `sequence`
+in the same way as v1.0. (`sequence` may become a core attribute with a particular meaning.)
+The proxy could make a choice of:
+
+- Expecting that the previous attribute *is* used in the same way, and treating
+  it as per v1.0 events, with a suitable failure mode.
+- Not attaching any significance to the attribute for v1.1 events, treating it as
+  any other extension attribute.
+
+There's no one-size-fits-all approach here.
```

Comment:
I wonder what happens for proxies that do things like convert binary CEs to structured CEs, and in the process upgrade the version string?